### PR TITLE
Turn into module

### PR DESCRIPTION
**Pull request by nikku**
This allows us to consume the library as an ES module.

----

If we had this, then we could likely do the following:

```html
<html>
  <script type="module">
    import Viewer from 'https://unpkg.com/bpmn-js/@16.0.0/lib/Viewer.js';
    
    const viewer = new Viewer();
  </script>
</html>
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,13 +2,14 @@
   "name": "path-intersection",
   "version": "2.2.1",
   "description": "Computes the intersection between two SVG paths",
-  "main": "intersect.js",
+  "module": "intersect.js",
```
Comment (barmac) [inline package.json]: Do we need this? According to the docs, `main` is the field name for entry point: https://docs.npmjs.com/cli/v10/configuring-npm/package-json#main

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,10 @@ All notable changes to [path-intersection](https://github.com/bpmn-io/path-inter
 
 ___Note:__ Yet to be released changes appear here._
 
+## 3.0.0
+
+* `FEAT`: turn into module
```
Comment (barmac) [inline CHANGELOG.md:11]: Breaking changes missing. We could copy from https://github.com/bpmn-io/bpmn-js/blob/develop/CHANGELOG.md#100

File: package.json (hunk 1 above)
Comment (barmac) [reply package.json]:
```suggestion
  "main": "intersect.js",
```

File: CHANGELOG.md (hunk 2 above)
Comment (barmac) [reply CHANGELOG.md:11]:
```suggestion
* `FEAT`: turn into module

### Breaking Changes

* You must now configure a module transpiler such as Babel or Webpack in order to import from non-ES modules code.
```

File: package.json (hunk 1 above)
Comment (nikku) [reply package.json]: Let's try this, as the project itself is marked as `type="module"`.

However, this library is meant to be used in the browser, and it is a common convention to expose ES modules through the `module` field. 

Primary goal is to make bundlers happy. Usage in Node is only a secondary concern.

**Comment by nikku** [comment]
Updated the PR according to your comments @barmac.

**Review by barmac — APPROVED** [review]

**Comment by nikku** [comment]
I verified the following works :tada:

```html

<html>
  <head>
  </head>
  <body>
    <script type="module">
      import intersect from 'https://unpkg.com/path-intersection?module';

      var path0 = [ [ 'M', 0, 0 ], [ 'L', 100, 100 ] ];
      var path1 = 'M0,100L100,0';

      const intersection = intersect(path0, path1);
      // [ { x: ..., y: ..., segment1: ..., segment2: ... }, ... ]
      console.log(intersection);
    </script>
  </body>
</html>
```

**Pull request merged**